torch/utils/serialization/read_lua_file.py

```diff
@@ -475,9 +478,9 @@ def read_int(self):
 
     def read_long(self):
         if self.long_size is None:
-            return self._read('l')
+            return self._read(LONGLONG_TYPECODE)
         elif self.long_size is 8:
-            return self._read('q')
+            return self._read(LONGLONG_TYPECODE)
```

Comment:
That's not the whole point of this `if`. If it's not set, you should fall back to native size